### DISCUSSION TITLE
Getting error when replaying past events

### DISCUSSION THREAD

**Discussion by vikasguptadev** [spatie/laravel-event-sourcing#448] [Q&A] [answered]
Hi,

I am following the tutorial and I am stuck on the topic when trying to create my second projector TransactionCountProjector

![image](https://github.com/spatie/laravel-event-sourcing/assets/57398077/7d459515-3c86-4534-b49d-d65fcad87f54)

I have created DB table, projection and projector. But when I am trying to replay the past events, I am getting error:

```
The `App\Models\TransactionCount` projection is not writeable at this point, please call `$model->writeable()` first.
```

which is referring to below line:

```
$tc = TransactionCount::firstOrCreate(['account_uuid' => $event->uuid]);
```

My code for projector is:

```
<?php

namespace App\Projectors;

use App\Events\MoneyAdded;
use App\Events\MoneySubtracted;
use App\Models\TransactionCount;
use Spatie\EventSourcing\EventHandlers\Projectors\Projector;

class TransactionCountProjector extends Projector
{
    public function onMoneyAdded(MoneyAdded $event)
    {
        $tc = TransactionCount::firstOrCreate(['account_uuid' => $event->uuid]);
        $tc->count += 1;
        $tc->writeable()->save();
    }

    public function onMoneySubtracted(MoneySubtracted $event){
        $tc = TransactionCount::firstOrCreate(['account_uuid' => $event->uuid]);
        $tc->count -= 1;
        $tc->writeable()->save();
    }
}
```

**Comment by sebastiandedeyne** [comment]
A projection must always be explicitly marked as writeable, also when created. Try `(new TransactionCount())->writeable()->firstOrCreate(…)` in the first lines of the projector methods.

**Reply by vikasguptadev** [reply]
@sebastiandedeyne 

I added `writeable()` and error is not coming anymore. So I think we need to update the code snippet on the website 

https://spatie.be/docs/laravel-event-sourcing/v7/using-projectors/writing-your-first-projector

![image](https://github.com/spatie/laravel-event-sourcing/assets/57398077/89f33def-9637-48aa-9ec5-249244e6bdd2)

**Reply by vikasguptadev** [reply]
However, I am getting another error now which seems to be unrelated to this one:

When replaying the events by executing the command `artisan event-sourcing:replay` I am getting error `artisan event-sourcing:replay`

But when I list the projectors with the command `artisan event-sourcing:list`, it is being listed there

![image](https://github.com/spatie/laravel-event-sourcing/assets/57398077/f58f0d93-5515-4fd1-ad8c-6a79a42d0ac4)

**Reply by vikasguptadev** [reply]
This is still not resolved. When I try to run the projector by firing off new event, I am getting below error

![image](https://github.com/spatie/laravel-event-sourcing/assets/57398077/54086fb1-10cc-41f9-babe-2ded6953f5fa)

`transactions_count` table is not supposed to have `uuid` column.

**Comment by vikasguptadev** [comment]
is there no solution for this problem?

**Reply by vikasguptadev** [reply, marked as answer]
Anyone who is looking for the answer:

I resolved this by adding a public method to my TransactionCount Model

```
public function getKeyName()
    {
        return 'account_uuid';
    }
```

The `Projection` wrapper interfere whenever you try to save the model. By default, it considers the uuid column name as `uuid` But in my case, it was named as `account_uuid`. So by using that method, I instructed the Projection to use `account_uuid` as uuid name
